x/compute/internal/keeper/keeper.go

```diff
@@ -43,55 +55,145 @@ type Keeper struct {
+	authZPolicy   AuthorizationPolicy
+	//paramSpace    subspace.Subspace
 }
 
 // NewKeeper creates a new contract Keeper instance
 // If customEncoders is non-nil, we can use this to override some of the message handler, especially custom
-func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, accountKeeper auth.AccountKeeper, bankKeeper *bank.Keeper,
+func NewKeeper(
+	cdc *codec.Codec,
+	storeKey sdk.StoreKey,
+	accountKeeper auth.AccountKeeper,
+	bankKeeper *bank.Keeper,
 	stakingKeeper *staking.Keeper,
-	router sdk.Router, homeDir string, wasmConfig types.WasmConfig, supportedFeatures string, customEncoders *MessageEncoders, customPlugins *QueryPlugins) Keeper {
+	router sdk.Router,
+	homeDir string,
+	wasmConfig types.WasmConfig,
+	supportedFeatures string,
+	customEncoders *MessageEncoders,
+	customPlugins *QueryPlugins,
+) Keeper {
 	wasmer, err := wasm.NewWasmer(filepath.Join(homeDir, "wasm"), supportedFeatures, wasmConfig.CacheSize)
 	if err != nil {
 		panic(err)
 	}
 
-	messenger := NewMessageHandler(router, customEncoders)
+	/*
+		// set KeyTable if it has not already been set
+		if !paramSpace.HasKeyTable() {
+			paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+		}
+	*/
 
 	keeper := Keeper{
 		storeKey:      storeKey,
 		cdc:           cdc,
 		wasmer:        *wasmer,
 		accountKeeper: accountKeeper,
 		bankKeeper:    *bankKeeper,
-		messenger:     messenger,
+		messenger:     NewMessageHandler(router, customEncoders),
 		queryGasLimit: wasmConfig.SmartQueryGasLimit,
+		// authZPolicy:   DefaultAuthorizationPolicy{},
+		//paramSpace:    paramSpace,
 	}
 	keeper.queryPlugins = DefaultQueryPlugins(bankKeeper, stakingKeeper, &keeper).Merge(customPlugins)
 	return keeper
 }
 
+/*
+func (k Keeper) getUploadAccessConfig(ctx sdk.Context) types.AccessConfig {
+	var a types.AccessConfig
+	k.paramSpace.Get(ctx, types.ParamStoreKeyUploadAccess, &a)
+	return a
+}
+
+func (k Keeper) getInstantiateAccessConfig(ctx sdk.Context) types.AccessType {
+	var a types.AccessType
+	k.paramSpace.Get(ctx, types.ParamStoreKeyInstantiateAccess, &a)
+	return a
+}
+
+// GetParams returns the total set of wasm parameters.
+func (k Keeper) GetParams(ctx sdk.Context) types.Params {
+	var params types.Params
+	k.paramSpace.GetParamSet(ctx, &params)
+	return params
+}
+
+func (k Keeper) setParams(ctx sdk.Context, ps types.Params) {
+	k.paramSpace.SetParamSet(ctx, &ps)
+}
+*/
+
 // Create uploads and compiles a WASM contract, returning a short identifier for the contract
 func (k Keeper) Create(ctx sdk.Context, creator sdk.AccAddress, wasmCode []byte, source string, builder string) (codeID uint64, err error) {
+	return k.create(ctx, creator, wasmCode, source, builder, &types.AccessConfig{Type: types.Everybody}, k.authZPolicy)
+}
+
+func (k Keeper) create(ctx sdk.Context, creator sdk.AccAddress, wasmCode []byte, source string, builder string, instantiateAccess *types.AccessConfig, authZ AuthorizationPolicy) (codeID uint64, err error) {
```

Comment:
This is super confusing. Any reason why we can't remove the stupid authorization stuff? Either way, don't use k.create vs k.Create

Reply:
the lowercase vs. uppercase thing is their design, so i'd rather not change it too much. as for the authorization stuff, we can probably remove it, but if that's the same as setting it to public, without removing it, then i'd rather keep it for the same merge resolution reasons as i mentioned above.